subsys/bluetooth/host/gatt.c

```diff
@@ -1438,6 +1438,27 @@ static void gatt_delayed_store_enqueue(uint8_t id, const bt_addr_le_t *peer_addr
 	}
 }
 
+static void do_delayed_store(uint8_t id, const bt_addr_le_t *peer_addr)
```

Comment:
Could `do_delayed_store` take `struct ds_peer *el` directly instead of `id and `peer_addr`? It seems like some work is done twice here to find `el`.

Reply:
Yes, I have done that because I would like to remove the condition on `CONFIG_BT_SETTINGS_DELAYED_STORE` in `bt_gatt_disconnected` and only use `do_delayed_store`. It's for a future work, I can use a `struct ds_peer *el` for now if you want.

Reply:
If you are planning to change things here anyway you can keep it as is.